doc/rtd/topics/network-config-format-v2.rst

```diff
@@ -177,6 +177,72 @@ Enable DHCP for IPv4. Off by default.
 
 Enable DHCP for IPv6. Off by default.
 
+**dhcp4-overrides**: *<(mapping)>*
```

Comment:
The following diff simplifies and improves the documentation. By:

* Not documenting the sub-properties, but referring to netplan's doc.
* Grouping dhcp4-overrides and dhcp6-overrides in one section.
* Clarify that with v2 netconfs and netplan as renderer, the overrides are also applied.

Please, feel free to modify/rephrase it.

```diff
diff --git a/doc/rtd/topics/network-config-format-v2.rst b/doc/rtd/topics/network-config-format-v2.rst
index 8597cc32b..b00d4110e 100644
--- a/doc/rtd/topics/network-config-format-v2.rst
+++ b/doc/rtd/topics/network-config-format-v2.rst
@@ -8,6 +8,8 @@ version 2 format defined for the `netplan`_ tool.  Cloud-init supports
 both reading and writing of Version 2; the latter support requires a
 distro with `netplan`_ present.
 
+.. _Netplan Passthrough:
+
 Netplan Passthrough
 -------------------
 
@@ -177,62 +179,32 @@ Enable DHCP for IPv4. Off by default.
 
 Enable DHCP for IPv6. Off by default.
 
-**dhcp4-overrides**: *<(mapping)>*
-
-Some DHCP behaviors can be overridden. Overrides will only have an effect if
-the corresponding DHCP type is enabled. Optional.
-
-``hostname`` *<(scalar)>* sets the hostname which is sent to the DHCP server,
-instead of the machine's hostname. Optional.
-``route-metric`` *<(scalar)>* sets the default metric for routes that are
-automatically added. This allows prioritizing routes for devices. Optional.
-``send-hostname`` *<(bool)>* when ``true``, the machine's hostname will be sent
-to the DHCP server. Default: ``true``. Optional.
-``use-dns`` *<(bool)>* when ``true``, the DNS servers received from DHCP will
-be used and take precedence over configured ones. Default: ``true``. Optional.
-``use-hostname`` *<(bool)>* when ``true``,  the hostname received from DHCP
-will be set as the transient hostname of the machine. Default: ``true``. Optional.
-``use-mtu`` *<(bool)>* when ``true``, the MTU received from DHCP will be used
-as the MTU for the network interface. Default: ``true``. Optional.
-``use-ntp`` *<(bool)>* when ``true``, the NTP servers received from DHCP will
-be used by systemd-timesyncd and take precendence over configured ones.
-Default: ``true``. Optional.
-``use-domains`` *<(scalar)>* takes a boolean or the value ``route``. When
-``true``, the domain name received from DHCP will be used as the DNS search
-domain for this network interface. When ``route``, the domain name received
-from DHCP will be used for routing DNS only, not for searching. Optional.
-
-NOTE: This is currently only supported for the ``networkd`` renderer. As of
-writing this the only known usage of this renderer is with the VMware data
-source.
+**dhcp4-overrides** and **dhcp6-overrides**: *<(mapping)>*
 
-Example: ::
+DHCP behavior overrides. Overrides will only have an effect if
+the corresponding DHCP type is enabled. Refer to `netplan#dhcp-overrides`_
+for more documentation.
 
-  dhcp4-overrides:
-    hostname: hal
-    route-metric: 1100
-    send-hostname: false
-    use-dns: false
-    use-domains: false
-    use-hostname: false
-    use-mtu: false
-    use-ntp: false
-    use-routes: false
-
-**dhcp6-overrides**: *<(mapping)>*
+.. note::
 
-Some DHCP behaviors can be overridden. Overrides will only have an effect if
-the corresponding DHCP type is enabled. Optional.
+  These properties are only consumed on ``netplan`` and ``networkd``
+  renderers.
 
-See ``dhcp4-overrides`` above for an explanation of the fields.
+The ``netplan`` renderer :ref:`passes through <Netplan Passthrough>`
+everything and the ``networkd`` renderer consumes the following sub-properties:
 
-NOTE: This is currently only supported for the ``networkd`` renderer. As of
-writing this the only known usage of this renderer is with the VMware data
-source.
+* ``hostname``
+* ``route-metric``
+* ``send-hostname``
+* ``use-dns``
+* ``use-domains``
+* ``use-hostname``
+* ``use-mtu``
+* ``use-ntp``
 
 Example: ::
 
-  dhcp6-overrides:
+  dhcp4-overrides:
     hostname: hal
     route-metric: 1100
     send-hostname: false
@@ -593,4 +565,5 @@ This is a complex example which shows most available features: ::
         dhcp4: yes
 
 .. _netplan: https://netplan.io
+.. _netplan#dhcp-overrides: https://netplan.io/reference#dhcp-overrides
 .. vi: textwidth=79

```

Reply:
Thanks! This patch looks great!